File: Queue/Queue.java
import java.util.Scanner;

class node
{
    public int data;
    public node next;
}

class QueueX
{
    private node first;
    private int iCount;

    public QueueX()
    {
        this.first = null;
        this.iCount = 0;
    }

    public void enqueue(int no)             
    {
        node temp = null;
        node newn = new node();

        newn.data = no;
        newn.next = null;

        if(this.first == null)
        {
            this.first = newn;
        }
        else
        {
            temp = this.first;

            while(temp.next != null)
            {
                temp = temp.next;
            }

            temp.next = newn;
        }
        iCount++;
    }

    public int dequeue()           
    {
        node temp = first;

        if(this.first == null)
        {
            System.out.println("Unable to remove as Queue is empty");
            return -1;
        }
        else
        {
            first = first.next;
            iCount--;
            return temp.data;
        }
    }

    public void Display()
    {
        node temp = null;

        temp = this.first;

        while(temp != null)
        {
            System.out.println("| "+temp.data+" | - ");
            temp = temp.next;
        }
        System.out.println();
    }

    public int Count()
    {
        return iCount;
    }
}

class Queue
{
    public static void main(String A[])
    {
        QueueX sobj = new QueueX();
        Scanner sobj1 = new Scanner(System.in);
        int iRet = 0, iValue = 0, iChoice = 0;

        System.out.print("-------------------------------------------------------------------\n");
        System.out.print("---------------------------------Queue------------------------------\n");
        System.out.print("-------------------------------------------------------------------\n\n");

        while(true)
        {
            System.out.print("-------------------------------------------------------------------\n");
            System.out.print("----------------------Please Select the option----------------------\n");
            System.out.print("-------------------------------------------------------------------\n");

            System.out.print("1 : Insert new node in the Queue\n");
            System.out.print("2 : Remove Element from the Queue\n");
            System.out.print("3 : Display all elements from the Queue\n");
            System.out.print("4 : Count all the elements from the Queue\n");
            System.out.print("0 : Terminate the position\n");
            System.out.print("-------------------------------------------------------------------\n");

            iChoice = sobj1.nextInt();

            if(iChoice == 1)
            {
                System.out.print("Enter the data that you want to insert : \n");
                iValue = sobj1.nextInt();
                sobj.enqueue(iValue);
            }
            else if(iChoice == 2)
            {
                System.out.print("Remove the element from the stack\n");
                iRet = sobj.dequeue();
                System.out.println("Removed element from the Queue is : \n"+iRet);
            }
            else if(iChoice == 3)
            {
                sobj.Display();
            }
            else if(iChoice == 4)
            {
                iRet = sobj.Count();
                System.out.print("The number of elements in the Queue are : "+iRet);
            }
            else if(iChoice == 0)
            {
                System.out.print("Thank you for using our application\n");
                break;
            }
            else
            {
                System.out.print("Invalid choice\n");
            }
            System.out.print("\n-------------------------------------------------------------------\n");
        }
    }
}

File: Queue/Queue.c
#include<stdio.h>
#include<stdlib.h>

struct node 
{
    int data;
    struct node *next;
};

typedef struct node NODE;
typedef struct node *PNODE;
typedef struct node **PPNODE;

void enqueue(PPNODE first, int no)
{
    PNODE newn = NULL;
    PNODE temp = NULL;

    newn = (PNODE)malloc(sizeof(NODE));

    newn->data = no;
    newn->next = NULL;

    if((*first) == NULL)
    {
        (*first) = newn;
    }
    else
    {
        temp = *first;

        while(temp->next != NULL)
        {
            temp = temp->next;
        }

            temp->next = newn;
        }
}

void Display(PNODE first)
{
    while(first != NULL)
    {
        printf("| %d |\n ",first->data);
        first = first -> next;
    }
    printf("\n");
}

int Count(PNODE first)
{
    int iCount = 0;

    while(first != NULL)
    {
        iCount++;
        first = first -> next;
    }
    return iCount;
}

int dequeue(PPNODE first)
{
    PNODE temp = *first;

    if(*first == NULL)
    {
        printf("Unable to remove as Queue is empty\n");
        return -1;
    }
    else
    {
        *first = (*first)->next;
        return temp->data;
    }
}

int main()
{
    PNODE head = NULL;
    int iChoice = 0, iValue = 0, iRet = 0;

    printf("-------------------------------------------------------------------\n");
    printf("---------------------------------Queue-----------------------------\n");
    printf("-------------------------------------------------------------------\n\n");

    while(1)
    {
        printf("-------------------------------------------------------------------\n");
        printf("----------------------Please Select the option----------------------\n");
        printf("-------------------------------------------------------------------\n");

        printf("1 : Insert new node in the Queue\n");
        printf("2 : Remove element from the Queue\n");
        printf("3 : Display all elements from the Queue\n");
        printf("4 : Count all the elements from the Queue\n");
        printf("0 : Terminate the position\n");
        printf("-------------------------------------------------------------------\n");

        scanf("%d",&iChoice);

        if(iChoice == 1)
        {
            printf("Enter the data that you want to insert : \n");
            scanf("%d",&iValue);
            enqueue(&head,iValue);
        }
        else if(iChoice == 2)
        {
            printf("Remove the element from the Queue\n");
            iRet = dequeue(&head);
            printf("Removed element from the Queue is : %d\n",iRet);
        }
        else if(iChoice == 3)
        {
            Display(head);
        }
        else if(iChoice == 4)
        {
            iRet = Count(head);
            printf("The number of elements in the Queue are : %d\n",iRet);
        }
        else if(iChoice == 0)
        {
            printf("Thank you for using our application\n");
            break;
        }
        else
        {
            printf("Invalid choice\n");
        }
        printf("\n-------------------------------------------------------------------\n");
    }

    return 0;
}